scipy/optimize/_differentialevolution.py

```diff
@@ -434,6 +439,13 @@ def init_population_random(self):
         rng = self.random_number_generator
         self.population = rng.random_sample(self.population_shape)
 
+        # reset population energies
+        self.population_energies = (np.ones(self.num_population_members)
+                                    * np.inf)
+
+        # reset number of function evaluations counter
+        self._nfev = 0
+
```

Comment:
In retrospect it could've been better to have a single method which branches:

    def init_population(self, init='latinhypercube'):
        if init == 'latinhypercube':
            pass
        if init == 'random':
            pass

etc. There are currently duplicate lines of code across the two methods because when the populations are initialised `_nfev` and `population_energies` need to be initialised as well. This could be done in `__init__`. However, I was trying to cover the case where someone reuses the object. For example:

    solver = DifferentialEvolutionSolver(func, bounds, init='latinhypercube')
    solver.solve()
    #reuse solver, and re-init population
    solver.init_population_random()
    solver.solve()